### PR TITLE
Testing Math Magician

### PR DESCRIPTION
**Pull request by kbjude**
- Unit tests for Math Magician
- Adding the other pages of Math Magician
- Working with Routes
- and many others

### CODE REVIEW

**Comment by bellom** [comment]
> * Unit tests for Math Magician
> * Adding the other pages of Math Magician
> * Working with Routes
> * and many others

@kbjude 
Please you need to be specific when writing a PR description, we both know `many others` don't really point to any thing.
kindly take note and update this PR description.
Thank you.

File: src/components/ButtonPanel.js [hunk 1]
```diff
@@ -1,43 +1,46 @@
...
+          <Button name="-" clickHandler={handleClick} />
+        </div>
+        <div className="row4">
+          <Button name="1" clickHandler={handleClick} />
+          <Button name="2" clickHandler={handleClick} />
+          <Button name="3" clickHandler={handleClick} />
+          <Button name="+" clickHandler={handleClick} />
+        </div>
+        <div className="row5">
+          <Button name="0" className="button-0" clickHandler={handleClick} />
```
Comment (bellom) [inline src/components/ButtonPanel.js:38]: ### [Optional]

While we focus on testing in this project, we can also make your project better.
Kindly give the `button 0` 50% `width` instead `25%`.

**With 25% width which is your currently width;**
image

**With 50% width;**
image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @kbjude :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Your tests successfully assure that react components are being properly displayed. However, as said in the requirements we also need to test the calculator functionalities. We need to write tests to verify if each operation (addition, subtraction, multiplication...) and buttons are working as expected.**

image

**Add unit tests for your calculate.js file; that is cover all the functionality in `calculate.js` with unit testing**.

### [Optional]
_You might want to work on your file structure, I don't think you really need two `app.js`_ 

_**Kindly check out my comments below;**_

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by christianotieno — CHANGES_REQUESTED** [review]
## Changes Required ♻️ 
Hello @kbjude 👋🏾 
You have done a great job with the recent changes after the review. The test suites coverage are pretty solid. 👌🏾 Keep it up.

### Change
The last review had three main requirements; two mandatory and one optional. Note that we should always aim to have our repo looking professional at all times, and the PR messages make part of this. They should stand out as well, and they are often a magnet for keen recruiters.  I urge us to improve the PR message for the branch for clarity, especially on the changes the feature branch proposes to add to the main codebase.  

Let's work on this before the project gets approved.

### Happy coding 💻 
#### Christian Otieno (CR)

**Comment by kbjude** [comment]
Testing the:
- addition
- Multiplication
- Division functionality in response to the TSE's feedback

**Review by wathigo — APPROVED** [review]
## Approved ✔️
Great job implementing the suggested changes.

**You may move on.** 

**Happy Coding 💻 
Simon Wathigo**

**Pull request merged**